en: add handle-alerts.md and alert-rules.md (#132)
pingcap/docs-dm: en/handle-alerts.md

---
title: Alert Handling
summary: Understand the handling methods of major alert information in DM.
category: reference
---

# Alert Handling

This document introduces how to deal with the major alert information in DM.

## The alert of task status

### `DM_task_state`

The alert triggered when the DM-worker has sub-task in `Paused` status for over 20 minutes. You need to refer to [DM error handling](error-handling.md#Data-Migration-Error-Handling).

> ## The alert of relay log

Reviewer: ```suggestion
## Alert rules related to relay log
```

> ### `DM_relay_process_exits_with_error`

Reviewer: Please update other alert rules accordingly. `When something happens, an alert is triggered `~
Please update changes in `zh/monitor-a-dm-cluster.md` & `zh/replicate-data-using-dm.md` in https://github.com/pingcap/docs-dm/pull/93/files, thanks~

When the relay log processing unit has some error, an alert is triggered and the relay log processing unit's status becomes to `Paused`. You need to refer to [DM error handling](error-handling.md#Data-Migration-Error-Handling).

### `DM_remain_storage_of_relay_log`

When the disk free space of relay log was less than 10G, an alert is triggered. You can refer the handling method below.

- Delete unwanted data by manual to increase disk free space.

> - Try to adjust the [automatic data purge policy of relay log](relay-log.md#Automatic data purge) or execute [manual data purge](relay-log.md#Manual-data-purge).

Reviewer: ```suggestion
- Reconfigure the [automatic data purge strategy of the relay log](relay-log.md#automatic-data-purge) or[purge data manually](relay-log.md#manual-data-purge).
```

> - Try to [migrate DM-worker instance](cluster-operations.md#Replace/migrate-a-DM-master-instance) to another disk which has enough free space.

Reviewer: ```suggestion
- [Migrate the DM-worker instance](cluster-operations.md#replacemigrate-a-dm-worker-instance) to a disk with enough free space.
```

### `DM_fail_to_read_binlog_from_master`

When some error occurs while the relay log processing unit tries to read binlog event from upstream, an alert is triggered immediately and the relay log processing unit's status becomes to `Paused`. You need to refer to [DM error handling](error-handling.md#Data-Migration-Error-Handling).

### `DM_fail_to_write_relay_log`

When some error occurs while the relay log processing unit tries to write binlog event into the relay log file, an alert is triggered immediately and the relay log processing unit's status becomes to `Paused`. You need to refer to [DM error handling](error-handling.md#Data-Migration-Error-Handling).

### `DM_binlog_file_gap_between_master_relay`

When the number of latest binlog files extracted by the relay log processing unit lags behind the current upstream MySQL/MariaDB number by 1 (excluding 1) for over 10 minutes, an alert is triggered. You need to refer to [DM error handling](error-handling.md#Data-Migration-Error-Handling).

## Alert rules related to Dump/Load

### `DM_dump_process_exists_with_error`

> When the dump processing unit has some error, an alert is triggered and the dump processing unit'status becomes to `Paused`. You need to refer to [DM error handling](error-handling.md#Data-Migration-Error-Handling).

Reviewer: ```suggestion
When the Dump processing unit encounters an error, an alert is triggered and this unit moves to the `Paused` status. You need to refer to [Troubleshoot DM](error-handling.md#troubleshooting).
```

### `DM_load_process_exists_with_error`

> When the load processing unit has some error, an alert is triggered and the load processing unit'status becomes to `Paused`. You need to refer to [DM error handling](error-handling.md#Data-Migration-Error-Handling).

Reviewer: ```suggestion
When the Load processing unit encounters an error, an alert is triggered and this unit moves to the `Paused` state. You need to refer to [Troubleshoot DM](error-handling.md#troubleshooting).
```

> ## Aler rules related to binlog replication

Reviewer: ```suggestion
## Alert rules related to binlog replication
```
